### PR TITLE
Add reserved type, fixes #4466

### LINKED ISSUE #4466: VTOL transition switch broken in current master
It appears to be related to the radio settings
Transition switch is ignored (quad spins up with throttle when switch on)
Also, the pusher ESC does not initialize, or only does so after changing the VT_IDLE_PWM_MC, it does not seem to matter to what direction to change this.

@AndreasAntener @tumbili @LorenzMeier

### PR DESCRIPTION
**Pull request by sanderux**

### CODE REVIEW

File: src/modules/commander/commander_helper.cpp [hunk 1]
```diff
@@ -88,13 +88,14 @@ bool is_multirotor(const struct vehicle_status_s *current_status)
...
-	       || (current_status->system_type == MAV_TYPE_COAXIAL);
+		   || (current_status->system_type == MAV_TYPE_COAXIAL);
 }
 
 bool is_vtol(const struct vehicle_status_s * current_status) {
 	return (current_status->system_type == MAV_TYPE_VTOL_DUOROTOR ||
 		current_status->system_type == MAV_TYPE_VTOL_QUADROTOR ||
-		current_status->system_type == MAV_TYPE_VTOL_TILTROTOR);
+		current_status->system_type == MAV_TYPE_VTOL_TILTROTOR ||
+		current_status->system_type == MAV_TYPE_VTOL_RESERVED2);
```
Comment (LorenzMeier) [inline src/modules/commander/commander_helper.cpp]: Can you also add RESERVED3-5?

**Comment by AndreasAntener** [comment]
Before the refactoring we had `VEHICLE_TYPE_VTOL_OCTOROTOR` as enum for Quadplanes. Now VEHICLE_* type enum got removed, and we use the official one from mavlink where "our" quadplane type is equal to `MAV_TYPE_VTOL_RESERVED2`.

File: src/modules/commander/commander_helper.cpp (hunk 1 above)
Comment (sanderux) [inline src/modules/commander/commander_helper.cpp]: added

**Comment by LorenzMeier** [comment]
Nice!

**Pull request merged**